### PR TITLE
Clarify encoding of code challenge

### PR DESCRIPTION
**Pull request by mitchellhenke**
We had a partner that was having issues with mismatched code challenges, and the root cause was the encoding. Ruby's [urlsafe_encode64](https://ruby-doc.org/stdlib-2.4.0/libdoc/base64/rdoc/Base64.html#urlsafe_encode64-method) specifies that:
> This method complies with “Base 64 Encoding with URL and Filename Safe Alphabet” in RFC 4648. The alphabet uses ‘-’ instead of ‘+’ and ‘_’ instead of ‘/’.

This PR adds some clarification around the encoding and updates the Ruby script to be closer to the behavior within the IDP

### CODE REVIEW

**Review by russbeye — APPROVED** [review]
Great update to add some needed clarified based on our recent tech sync!

File: _pages/oidc.md [hunk 1]
```diff
@@ -133,12 +133,15 @@ https://idp.int.identitysandbox.gov/openid_connect/authorize?
   The unique identifier for the client. This will be registered with the Login.gov IdP in advance.
 
 * **code_challenge** — *required for PKCE*
-  The URL-safe base64 encoding of the SHA256 digest of a random value generated by the client. The original random value is referred to as the `code_verifier` and is later used with the token endpoint. Generating these values in Ruby might look like this, for example:
+  The URL-safe Base64 encoding of the SHA256 digest of a random value generated by the client. The original random value is referred to as the `code_verifier` and is later used with the token endpoint. Generating these values in Ruby might look like this, for example:
```
Comment (zachmargolis) [inline _pages/oidc.md:136]: Should we hyperlink to the RFC from this section?

File: _pages/oidc.md [hunk 2]
```diff
@@ -133,12 +133,15 @@ https://idp.int.identitysandbox.gov/openid_connect/authorize?
...
   ```ruby
   code_verifier = SecureRandom.hex
-  => "7a5e819dd39f17242fdeeba0c1c80be6"
+  => "5787d673fb784c90f0e309883241803d"
   code_challenge = Digest::SHA256.base64digest(code_verifier)
-  => "TdzfmaWefbtaI0Wdo6lrZCXpLu1WpamnSoSHfDUiL7Y="
+  => "1BUpxy37SoIPmKw96wbd6MDcvayOYm3ptT+zbe6L/zM="
+  url_safe_code_challenge = Base64.urlsafe_encode64(Base64.urlsafe_decode64(code_challenge))
```
Comment (zachmargolis) [inline _pages/oidc.md:142]: I feel like this example had a lot of hops. What if we updated it to do SHA256 digest as binary data and then use urlsafe base64 encoding of that?

**Review by zachmargolis — APPROVED** [review]
LGTM

File: _pages/oidc.md (hunk 2 above)
Comment (zachmargolis) [reply _pages/oidc.md:142]: so more like:
```ruby
  code_challenge = Digest::SHA256.digest(code_verifier) # binary data
  code_challenge.bytes
  => [212, 21, 41, 199, 45, 251, 74, 130, 15, 152, 172, 61, 235, 6, 221, 232, 192, 220, 189, 172, 142, 98, 109, 233, 181, 63, 179, 109, 238, 139, 255, 51]  
  Base64.urlsafe_encode64(code_challenge)
  => "1BUpxy37SoIPmKw96wbd6MDcvayOYm3ptT-zbe6L_zM="
```

File: _pages/oidc.md (hunk 1 above)
Comment (mitchellhenke) [reply _pages/oidc.md:136]: yep, can do

File: _pages/oidc.md (hunk 2 above)
Comment (mitchellhenke) [reply _pages/oidc.md:142]: For me, it was helpful to have the non-URL-safe encoding in the middle to see the differences between the steps.

Comment (zachmargolis) [reply _pages/oidc.md:142]: maybe we add a different "don't do this" example for the non-urlsafe at the end? I like the idea of having an easy-to-follow correct, happy path

```
Base64.encode64(code_challenge) # WRONG ENCODING
# => "1BUpxy37SoIPmKw96wbd6MDcvayOYm3ptT+zbe6L/zM=" # WRONG ENCODING
```

Comment (mitchellhenke) [reply _pages/oidc.md:142]: Sounds good 👍🏼

**Review by zachmargolis — APPROVED** [review]
LGTM!

File: _pages/oidc.md [hunk 3]
```diff
@@ -133,12 +133,16 @@ https://idp.int.identitysandbox.gov/openid_connect/authorize?
...
-  => "7a5e819dd39f17242fdeeba0c1c80be6"
-  code_challenge = Digest::SHA256.base64digest(code_verifier)
-  => "TdzfmaWefbtaI0Wdo6lrZCXpLu1WpamnSoSHfDUiL7Y="
+  => "5787d673fb784c90f0e309883241803d"
+  code_challenge = Digest::SHA256.digest(code_verifier) # binary data
+  url_safe_code_challenge = Base64.urlsafe_encode64(code_challenge)
+  # RFC 4648 URL-safe Base64 encoding replaces "+" with "-" and "/" with "_"
+  => "1BUpxy37SoIPmKw96wbd6MDcvayOYm3ptT-zbe6L_zM="
+  Base64.encode64(code_challenge) # wrong and URL-unsafe encoding
+  => "1BUpxy37SoIPmKw96wbd6MDcvayOYm3ptT+zbe6L/zM=" # wrong and URL-unsafe encoding
```
Comment (zachmargolis) [inline _pages/oidc.md:145]: one more idea... maybe rename the variables to match what's expted?

```suggestion
  digested_code_verifier = Digest::SHA256.digest(code_verifier) # binary data
  code_challenge = Base64.urlsafe_encode64(digested_code_verifier) # ✅ correct encoding
  # RFC 4648 URL-safe Base64 encoding replaces "+" with "-" and "/" with "_"
  => "1BUpxy37SoIPmKw96wbd6MDcvayOYm3ptT-zbe6L_zM="

  Base64.encode64(code_challenge) # ❌ wrong and URL-unsafe encoding
  => "1BUpxy37SoIPmKw96wbd6MDcvayOYm3ptT+zbe6L/zM=" # ❌ wrong and URL-unsafe encoding
```

**Pull request merged**